velox/docs/functions/spark/array.rst

```diff
@@ -27,6 +27,30 @@ Array Functions
 
         SELECT array_intersect(array(1, 2, 3), array(1, 3, 5)); -- [1,3]
 
+.. function:: array_max(array(E)) -> E
+
+    Returns maximum non-NULL element of the array. Returns NULL if array is empty or all elements are NULL.
+    When E is DOUBLE or REAL, returns NaN if any element is NaN. ::
+
+        SELECT array_max(ARRAY [1, 2, 3]); -- 3
+        SELECT array_max(ARRAY [-1, -2, -2]); -- -1
+        SELECT array_max(ARRAY [-1, -2, NULL]); -- -1
+        SELECT array_max(ARRAY []); -- NULL
+        SELECT array_max(ARRAY [Min, -0.0001, -0.0002, -0.0003, Max, kNaN]); -- NaN
```

Comment:
This is not a valid SQL since kNaN, Min and Max are not defined. Does Spark has a nan() function like Presto? If not, how NaN value is represented in Spark SQL? I assume this example doesn't need Min or Max, hence, these can be removed. Otherwise, please, change to actual values.

Reply:
OK, remove Min/Max, change NaN to float('nan')